token-swap/program/src/curve/constant_product.rs

```diff
@@ -34,53 +34,38 @@ pub struct ConstantProductCurve {
 
 impl CurveCalculator for ConstantProductCurve {
     /// Constant product swap ensures x * y = constant
-    fn swap(
+    fn swap_without_fees(
         &self,
         source_amount: u128,
         swap_source_amount: u128,
         swap_destination_amount: u128,
-    ) -> Option<SwapResult> {
-        // debit the fee to calculate the amount swapped
-        let trade_fee = self.trading_fee(source_amount)?;
-        let owner_fee = self.owner_trading_fee(source_amount)?;
-
-        let source_amount_less_fee = source_amount
-            .checked_sub(trade_fee)?
-            .checked_sub(owner_fee)?;
+    ) -> Option<SwapWithoutFeesResult> {
         let invariant = swap_source_amount.checked_mul(swap_destination_amount)?;
-        let mut new_source_amount_less_fee =
-            swap_source_amount.checked_add(source_amount_less_fee)?;
-        let mut new_destination_amount = invariant.checked_div(new_source_amount_less_fee)?;
+
+        let mut new_swap_source_amount = swap_source_amount.checked_add(source_amount)?;
+        let mut new_swap_destination_amount = invariant.checked_div(new_swap_source_amount)?;
+
         // Ceiling the destination amount if there's any remainder, which will
         // almost always be the case.
-        let remainder = invariant.checked_rem_euclid(new_source_amount_less_fee)?;
+        let remainder = invariant.checked_rem(new_swap_source_amount)?;
```

Comment:
Is there a reason to use `checked_rem` instead of `checked_rem_euclid`?

Reply:
Yeah, euclid is just for negative numbers

> Since, for the positive integers, all common definitions of division are equal, this is exactly equal to self % rhs.

https://doc.rust-lang.org/std/primitive.u32.html#method.rem_euclid